spec/td/command/query_spec.rb

```diff
@@ -23,20 +23,95 @@ module TreasureData::Command
       expect(client).to receive(:database).with('sample_datasets')
     end
 
-    it 'accepts --domain-key' do
-      expect(client).to receive(:query).
-        with("sample_datasets", "SELECT 1;", nil, nil, nil, {"domain_key"=>"hoge"}).
-        and_return(job)
-      op = List::CommandParser.new("query", %w[query], %w[], nil, ['--domain-key=hoge', '-dsample_datasets', 'SELECT 1;'], false)
-      command.query(op)
+    describe 'domain key' do
+      it 'accepts --domain-key' do
+        expect(client).to receive(:query).
+          with("sample_datasets", "SELECT 1;", nil, nil, nil, {"domain_key"=>"hoge"}).
+          and_return(job)
+        op = List::CommandParser.new("query", %w[query], %w[], nil, ['--domain-key=hoge', '-dsample_datasets', 'SELECT 1;'], false)
+        command.query(op)
+      end
+
+      it 'raises error if --domain-key is duplicated' do
+        expect(client).to receive(:query).
+          with("sample_datasets", "SELECT 1;", nil, nil, nil, {"domain_key"=>"hoge"}).
+          and_raise(::TreasureData::AlreadyExistsError.new('Query failed: domain_key has already been taken'))
+        op = List::CommandParser.new("query", %w[query], %w[], nil, ['--domain-key=hoge', '-dsample_datasets', 'SELECT 1;'], false)
+        expect{ command.query(op) }.to raise_error(TreasureData::AlreadyExistsError)
+      end
     end
 
-    it 'raises error if --domain-key is duplicated' do
-      expect(client).to receive(:query).
-        with("sample_datasets", "SELECT 1;", nil, nil, nil, {"domain_key"=>"hoge"}).
-        and_raise(::TreasureData::AlreadyExistsError.new('Query failed: domain_key has already been taken'))
-      op = List::CommandParser.new("query", %w[query], %w[], nil, ['--domain-key=hoge', '-dsample_datasets', 'SELECT 1;'], false)
-      expect{ command.query(op) }.to raise_error(TreasureData::AlreadyExistsError)
+    describe 'wait' do
+      let :job do
+        double('job', job_id: 123, update_status!: nil, debug: double.as_null_object)
+      end
+      before do
+        allow(job).to receive(:finished?){ [TreasureData::Job::STATUS_SUCCESS, TreasureData::Job::STATUS_ERROR, TreasureData::Job::STATUS_KILLED].include?(job.status) }
+        allow(job).to receive(:success?){ job.status == TreasureData::Job::STATUS_ERROR }
+        expect(client).to receive(:query).
+          with("sample_datasets", "SELECT 1;", nil, nil, nil, {}).
+          and_return(job)
+      end
+      it 'accepts --wait' do
+        now = 1_400_000_000
+        allow(Process).to receive(:clock_gettime).with(Process::CLOCK_MONOTONIC, :second){now}
+        allow(command).to receive(:sleep){|arg| now += arg}
+        allow(job).to receive(:status){ now > 1_400_000_010 ? TreasureData::Job::STATUS_SUCCESS : TreasureData::Job::STATUS_RUNNING }
+        op = List::CommandParser.new("query", %w[query], %w[], nil, ['--wait', '-dsample_datasets', 'SELECT 1;'], false)
+        command.query(op)
```

Comment:
`expect { command.query(op) }.not_to raise_error` is more clear what testing in this.